random1on1/api/channels.py

```diff
@@ -1,36 +1,132 @@
+import json
 from abc import ABC
 from abc import abstractmethod
+from datetime import datetime
+from functools import reduce
 
-from discord.channel import GroupChannel
-from discord.channel import TextChannel
+from discord import CategoryChannel
+from discord import GroupChannel
+from discord import Role
+from discord import TextChannel
+from networkx import connected_components
+from networkx import union
+from preconditions import preconditions
 
-from random1on1.api.participant import Participant
+from .pairings import Pairings
 
 
-class ReadWriteChannel(ABC, TextChannel):
+class Random1on1Channel(ABC):
 
+    @preconditions(lambda name: len(name) > 0)
     @abstractmethod
-    def __init__(self, channel_name: str):
-        # TODO: Add precondition checks here
-        self.channel_name = channel_name
+    def __init__(self, name: str, category: CategoryChannel):
+        self.name = name
+        channels = [c for c in category.text_channels if c.name == self.name]
+        if len(channels) == 0:
+            channel = category.create_text_channel(name=self.name)
+        elif len(channels) == 1:
+            channel = channels[0]
+        else:
+            raise RuntimeError(
+                f"Found multiple channels in category {category.name} with reserved channel name {self.name}"
+            )
 
+        if not isinstance(channel, TextChannel):
+            raise RuntimeError(
+                f"Channel {self.name} was found, but it is not a TextChannel")
```

Comment:
In `channels` you are only going through the text channels.

Reply:
Ah good point! I think I added this to deal with python type hinting issues (I originally wrote the code to look through all guild channels rather than the channels from the category, which had a different type signature). I'll look into removing it.